SEIRD parameterisation: made specific to COVID-19 (#54)
Como-DTC-Collaboration/como-models: vignettes/SEIRD.Rmd

## Using an SEIRD model to simulate COVID-19 dynamics
We now show how `comomodels` can be used to create an SEIRD which captures the dominant characteristics of an unmitigated COVID-19 epidemic.

We first create an `SEIRD` object that we will use to run simulations.
```{r}
my_model <- SEIRD()
```

The most crucial step in using compartmental models of transmission dynamics is their parameterisation because different parameter ranges allow both qualitatively and quantitatively different dynamics. There are two ways to estimate the parameters of such models: i) to use data collected in the laboratory or field to estimate individual parameters; or ii) to fit these models to field data and, in doing so, estimate potentially a collective of parameters. There are limitations to both approaches. The first does not account for how uncertainty in one parameter might influence our estimates of another: a higher value of $\beta$ can be offset by an increased value of $\mu$ to result in similar case counts, for instance. The second approach generally only allows a subset of parameters to be estimated, since the available data are sparse and insufficient to jointly determine all.

> Important though is that both approaches use data to determine parameters. And for practically all infectious diseases modelled, the available data do not pin down a model's parameters to a single set. It is thus important to determine whether the conclusions of modelling -- whether a new viral variant will lead a country's healthcare infrastructure to be overrun, for instance -- are sensitive to the parameter choices assumed. 

Reviewer: ```suggestion
However, what is most important is that both approaches use data to determine parameters. For practically all infectious diseases modelled, the available data do not pin down a model's parameters to a single set. Thus, it is important to determine whether the conclusions of modelling -- whether a new viral variant will lead a country's healthcare infrastructure to be overrun, for instance -- are sensitive to the parameter choices. 
```

Author: I've taken part of this and kept the other bit as was -- hope ok!

Here, we follow the first approach to parameterising the SEIRD model, and show later how a sensitivity analysis can be performed.

> The SEIRD model has four transmission parameters: $\beta$, $\kappa$, $\gamma$ and $\mu$. We choose $\kappa=1/5.5$, implicitly assuming a latency period of 5.5 days. We assume that 0.66\% of individuals infected with the virus die [@verity2020estimates]: this is termed an infection fatality ratio (IFR). Changing either $\gamma$ or $\mu$ affects both the rate at which individuals recover or die and the proportions which recover or die. As such, the implications of the individual parameter values are hard to intuit without recourse to the other. Here, we choose a different parameterisation that makes it simpler to set these parameters to appropriate values:

Reviewer: ```suggestion
The SEIRD model has four transmission parameters: $\beta$, $\kappa$, $\gamma$ and $\mu$. We choose $\kappa=1/5.5$, implicitly assuming a latency period of 5.5 days. We assume that 0.66\% of individuals infected with the virus die [@verity2020estimates]: this is termed an infection fatality ratio (IFR). Changing either $\gamma$ or $\mu$ affects both the rate at which individuals recover or die and the proportions of the population which recover or die. As such, the implications of the individual parameter values are difficult to divine without recourse to the other. Here, we choose a different parameterisation that makes it simpler to set these parameters to appropriate values:
```

Author: Sorry I prefer mine here!

\begin{align}
\gamma &= \zeta (1-\text{IFR})\\
\mu &= \zeta \text{IFR}.
\end{align}

The proportion of infecteds who go on to die due to infection is given by the ratio of the rate of death to the overall rate out from the infectious compartment, which is the definition of the IFR:

\begin{equation}
\frac{\zeta \text{IFR}}{\zeta \text{IFR} + \zeta (1-\text{IFR})} = \text{IFR}.
\end{equation}

The average duration spent in the infectious compartment is given by:

\begin{equation}
\frac{1}{\zeta \text{IFR} + \zeta (1-\text{IFR})} = \frac{1}{\zeta}.
\end{equation}

We assume an average duration of infectiousness of 2 days, which indicates that, after 10 days, <1\% would be infectious -- in line with UK government policy that assumes individuals cannot pass on infection after 10 days [@stayathome2021]. As such, we set $\zeta = 1/2$.

The remaining parameter, $\beta$, is set to a value to produce the same basic reproduction number as was assumed in the influential "report 9" from Imperial College [@ferguson2020report]: $R_0=2.4$ (see explanation of $R_0$ below).

```{r}
kappa <- 1 / 5.5

> IFR <- 0.0066

Reviewer: ```suggestion
load(file = "../data/infection_fatality_ratio.rda")

IFR <- infection_fatality_ratio$overall
```

Author: This now isn't needed at all with Solveig's covid_transmission_parameters function, which is great :)
